app/models/submission.rb

```diff
@@ -99,8 +102,28 @@ def steps
     STEPS.fetch(submission_type)
   end
 
-  def name
-    [I18n.l(created_at, format: :short), content.step01.try(:name)].compact.join(' ')
+  def object_name
+    case submission_type
+    when 'population' then content.step01.try(:name) || ''
+    when 'trial' then content.step01.try(:plant_trial_name) || ''
+    else ''
+    end
+  end
+
+  def object_description
+    case submission_type
+    when 'population' then content.step01.try(:description) || ''
+    when 'trial' then content.step01.try(:plant_trial_description) || ''
+    else ''
+    end
+  end
+
+  def affiliation
```

Comment:
This is actually not random, though it might look like it is. `object_` because it describe the object that was submitted, not the submission itself. Perhaps `submitted_object_name` would fit even better.

But `affiliation` is not something that describes the object. It is in fact the creator's affiliation. It is strange to think that a plant trial has an affiliation :). That's the whole reason.